Change default logging level to Info

amnk opened the pull request:
Recent PR (#1510) changed logging, and also introduced default logging
level. While it seemed like a good idea, it also introduced a lot of
confusion, because in logrus `Println()` is an alias to `Info()`
loglevel - so lots of messages were lost.

This change restores previous behavior in logging output.

Related: gruntwork-io/terragrunt#1529
Related: gruntwork-io/terragrunt#1530
Related: gruntwork-io/terragrunt#1531
Related: gruntwork-io/terragrunt#1532

brikis98 reviewed (commented):
Thanks! I think this will fix a lot of issues, but make the logging quite noisy again. Is there some easy way to get a list of all the places we log something, so I could make a call on which ones should be changed to debug, info, etc?

amnk commented:
@brikis98 just `grep` I'm afraid :). We can start with reviewing `grep 'logger.Printf' * -R`: `Printf()` is an alias to `Info()`, so all of those will be added

brikis98 commented:
> @brikis98 just `grep` I'm afraid :). We can start with reviewing `grep 'logger.Printf' * -R`: `Printf()` is an alias to `Info()`, so all of those will be added

Any chance you could do a grep and dump the list of log messages here? I can then quickly scan and see which should be changed to other log levels. I'm so booked right now, I don't even have time to figure out the right `grep` command to run!

yorinasub17 commented:
Here's what I got from `rg "(Info)|(Print)"` and cutting out the extraneous ones. I probably could get more context for some of these where I only got the function with `(`, but am also a bit buried to do more than this.

```
options/options.go
230:            logger.Printf("error: %v\n", errors.WithStackTrace(err))

util/retry.go
14:             logger.Printf(actionDescription)

config/dependency.go
472:    terragruntOptions.Logger.Infof("Detected module %s is already init-ed. Retrieving outputs directly from working directory.", targetConfig)

shell/ptty_windows.go
29:             options.Logger.Printf("failed to get console mode: %v\n", err)
34:             options.Logger.Printf("failed to set console mode: %v\n", err)

shell/ptty_unix.go
30:                     terragruntOptions.Logger.Printf("Error closing pty: %s", closeErr)
48:                             terragruntOptions.Logger.Printf("error resizing pty: %s", inheritSizeErr)
62:                     terragruntOptions.Logger.Printf("Error restoring terminal state: %s", restoreErr)
74:             terragruntOptions.Logger.Printf("Error forwarding stdin: %s", copyStdinErr)

shell/run_shell_cmd.go
60:     terragruntOptions.Logger.Infof("Running command: %s %s", command, strings.Join(args, " "))
177:                            logger.Printf("Forward signal %v to terraform.", s)
180:                                    logger.Printf("Error forwarding signal: %v", err)

shell/prompt.go
28:             terragruntOptions.Logger.Infof("The non-interactive flag is set to true, so assuming 'yes' for all prompts")

cli/hclfmt.go
32:             terragruntOptions.Logger.Printf("Formatting terragrunt.hcl file at: %s.", targetFile)
36:     terragruntOptions.Logger.Printf("Formatting terragrunt.hcl files from the directory tree %s.", terragruntOptions.WorkingDir)
69:     terragruntOptions.Logger.Printf("Formatting %s", tgHclFile)
73:             terragruntOptions.Logger.Printf("Error retrieving file info of %s", tgHclFile)
79:             terragruntOptions.Logger.Printf("Error reading %s", tgHclFile)
86:             terragruntOptions.Logger.Printf("Error parsing %s", tgHclFile)

dynamodb/dynamo_lock_table.go
47:             terragruntOptions.Logger.Printf("Lock table %s does not exist in DynamoDB. Will need to create it just this first time.", tableName)
84:     terragruntOptions.Logger.Printf("Creating table %s in DynamoDB", tableName)
103:                    terragruntOptions.Logger.Printf("Looks like someone created table %s at the same time. Will wait for it to be in active state.", tableName)
131:            terragruntOptions.Logger.Printf("No tags for lock table given.")
136:    terragruntOptions.Logger.Printf("Adding tags to lock table: %s", tags)
186:                    terragruntOptions.Logger.Printf("Success! Table %s is now in active state.", tableName)
191:            terragruntOptions.Logger.Printf("Table %s is not yet in active state. Will check again after %s.", tableName, sleepBetweenRetries)
206:            terragruntOptions.Logger.Printf("Table %s already has encryption enabled", tableName)
213:    terragruntOptions.Logger.Printf("Enabling server-side encryption on table %s in AWS DynamoDB", tableName)
225:                    terragruntOptions.Logger.Printf("Looks like someone is already updating table %s at the same time. Will wait for that update to complete.", tableName)
243:    terragruntOptions.Logger.Printf("Waiting for encryption to be enabled on table %s", tableName)
252:                    terragruntOptions.Logger.Printf("Encryption is now enabled for table %s!", tableName)
256:            terragruntOptions.Logger.Printf("Encryption is still not enabled for table %s. Will sleep for %v and try again.", tableName, sleepBetweenRetries)

cli/version_check.go
50:     terragruntOptions.Logger.Printf("Terraform version: %s", terraformVersion)

configstack/running_module.go
252:            module.Module.TerragruntOptions.Logger.Printf("Module %s has finished with an error: %v", module.Module.Path, moduleErr)

configstack/stack.go
80:             terragruntOptions.Logger.Println(output)
87:                             terragruntOptions.Logger.Printf("%v%v refers to remote state "+

cli/debug.go
24:     terragruntOptions.Logger.Infof(

cli/download_source.go
36:     terragruntOptions.Logger.Printf("Copying files from %s into %s", terragruntOptions.WorkingDir, terraformSource.WorkingDir)
43:     terragruntOptions.Logger.Printf("Setting working directory to %s", terraformSource.WorkingDir)
52:             terragruntOptions.Logger.Printf("The --%s flag is set, so deleting the temporary folder %s before downloading source.", OPT_TERRAGRUNT_SOURCE_UPDATE, terraformSource.DownloadDir)
64:             terragruntOptions.Logger.Printf("Terraform files in %s are up to date. Will not download again.", terraformSource.WorkingDir)
107:            terragruntOptions.Logger.Printf("Working dir %s exists but contains no Terraform files, so assuming code needs to be downloaded again.", terraformSource.WorkingDir)
149:    terragruntOptions.Logger.Printf("Downloading Terraform configurations from %s into %s", terraformSource.CanonicalSourceURL, terraformSource.DownloadDir)

cli/cli_app.go
357:            terragruntOptions.Logger.Infof(
537:    terragruntOptions.Logger.Printf("Detected %d Hooks", len(hooks))
542:                    terragruntOptions.Logger.Printf("Executing hook: %s", curHook.Name)
548:                            terragruntOptions.Logger.Printf("Error running hook %s with message: %s", curHook.Name, possibleError.Error())
677:            terragruntOptions.Logger.Printf("Errors encountered running before_hooks. Not running '%s'.", description)
711:                            terragruntOptions.Logger.Printf("Encountered an error eligible for retrying. Sleeping %v before retrying.\n", terragruntOptions.Sleep)
925:    terragruntOptions.Logger.Infof("%s", stack.String())

remote/remote_state.go
67:     terragruntOptions.Logger.Printf("Initializing remote state for the %s backend", remoteState.Backend)
110:            terragruntOptions.Logger.Infof("Backend type has changed from %s to %s", existingBackend.Type, remoteState.Backend)
119:    terragruntOptions.Logger.Printf("Backend %s has not changed.", existingBackend.Type)

remote/remote_state_s3.go
128:            terragruntOptions.Logger.Printf("%s\n", lockTableDeprecationMessage)
179:            terragruntOptions.Logger.Printf("Backend type has changed from s3 to %s", existingBackend.Type)
196:                    terragruntOptions.Logger.Printf("Remote state configuration encrypt contains invalid value %v, should be boolean.", existingBackend.Config["encrypt"])
243:            terragruntOptions.Logger.Printf("%s\n", lockTableDeprecationMessage)
337:            terragruntOptions.Logger.Printf("WARNING: encryption is not enabled on the S3 remote state bucket %s. Terraform state files may contain secrets, so we STRONGLY recommend enabling encryption!", config.Bucket)
383:            terragruntOptions.Logger.Printf("WARNING: Versioning is not enabled for the remote state S3 bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
395:                    terragruntOptions.Logger.Printf("Looks like you're already creating bucket %s at the same time. Will not attempt to create it again.", config.remoteStateConfigS3.Bucket)
407:            terragruntOptions.Logger.Printf("Root access is disabled for the remote state S3 bucket %s using 'skip_bucket_root_access' config.", config.remoteStateConfigS3.Bucket)
413:            terragruntOptions.Logger.Printf("TLS enforcement is disabled for the remote state S3 bucket %s using 'skip_bucket_enforced_tls' config.", config.remoteStateConfigS3.Bucket)
427:            terragruntOptions.Logger.Printf("Versioning is disabled for the remote state S3 bucket %s using 'skip_bucket_versioning' config.", config.remoteStateConfigS3.Bucket)
433:            terragruntOptions.Logger.Printf("Server-Side Encryption is disabled for the remote state AWS S3 bucket %s using 'skip_bucket_ssencryption' config.", config.remoteStateConfigS3.Bucket)
439:            terragruntOptions.Logger.Printf("___WARNING___: The terragrunt configuration option 'skip_bucket_accesslogging' is now deprecated. Access logging for the state bucket %s is disabled by default. To enable access logging for bucket %s, please provide property `accesslogging_bucket_name` in the terragrunt config file. For more details, please refer to the Terragrunt documentation.", config.remoteStateConfigS3.Bucket, config.remoteStateConfigS3.Bucket)
443:            terragruntOptions.Logger.Printf("Enabling bucket-wide Access Logging on AWS S3 bucket %s - using as TargetBucket %s", config.remoteStateConfigS3.Bucket, config.AccessLoggingBucketName)
446:                    terragruntOptions.Logger.Printf("Error: Could not create logs bucket %s for AWS S3 bucket %s", config.AccessLoggingBucketName, config.remoteStateConfigS3.Bucket)
454:            terragruntOptions.Logger.Printf("Access Logging is disabled for the remote state AWS S3 bucket %s", config.remoteStateConfigS3.Bucket)
478:            terragruntOptions.Logger.Printf("No tags specified for bucket %s.", config.remoteStateConfigS3.Bucket)
485:    terragruntOptions.Logger.Printf("Tagging S3 bucket with %s", config.S3BucketTags)
497:    terragruntOptions.Logger.Printf("Tagged S3 bucket with %s", config.S3BucketTags)
519:    terragruntOptions.Logger.Printf("Waiting for bucket %s to be created", config.Bucket)
522:                    terragruntOptions.Logger.Printf("S3 bucket %s created.", config.Bucket)
525:                    terragruntOptions.Logger.Printf("S3 bucket %s has not been created yet. Sleeping for %s and will check again.", config.Bucket, SLEEP_BETWEEN_RETRIES_WAITING_FOR_S3_BUCKET)
535:    terragruntOptions.Logger.Printf("Creating S3 bucket %s", aws.StringValue(bucket))
540:    terragruntOptions.Logger.Printf("Created S3 bucket %s", aws.StringValue(bucket))
553:    terragruntOptions.Logger.Printf("Enabling root access to S3 bucket %s", config.Bucket)
593:    terragruntOptions.Logger.Printf("Enabled root access to bucket %s", config.Bucket)
599:    terragruntOptions.Logger.Printf("Enabling enforced TLS access for S3 bucket %s", config.Bucket)
635:    terragruntOptions.Logger.Printf("Enabled enforced TLS access for bucket %s", config.Bucket)
641:    terragruntOptions.Logger.Printf("Enabling versioning on S3 bucket %s", config.Bucket)
652:    terragruntOptions.Logger.Printf("Enabled versioning on S3 bucket %s", config.Bucket)
658:    terragruntOptions.Logger.Printf("Enabling bucket-wide SSE on AWS S3 bucket %s", config.Bucket)
671:    terragruntOptions.Logger.Printf("Enabled bucket-wide SSE on AWS S3 bucket %s", config.Bucket)
682:    terragruntOptions.Logger.Printf("Putting bucket logging on S3 bucket %s with TargetBucket %s and TargetPrefix %s", config.Bucket, logsBucket, targetPrefix)
698:    terragruntOptions.Logger.Printf("Enabled bucket-wide Access Logging on AWS S3 bucket %s", config.Bucket)
706:    terragruntOptions.Logger.Printf("Blocking all public access to S3 bucket %s", config.Bucket)
723:    terragruntOptions.Logger.Printf("Blocked all public access to S3 bucket %s", config.Bucket)
731:    terragruntOptions.Logger.Printf("Granting WRITE and READ_ACP permissions to S3 Log Delivery (%s) for bucket %s. This is required for access logging.", s3LogDeliveryGranteeUri, aws.StringValue(bucket))
748:    terragruntOptions.Logger.Printf("Waiting for ACL bucket %s to have the updated ACL for access logging.", aws.StringValue(bucket))
774:                    terragruntOptions.Logger.Printf("Bucket %s now has the proper ACL permissions for access logging!", aws.StringValue(bucket))
778:            terragruntOptions.Logger.Printf("Bucket %s still does not have the ACL permissions for access logging. Will sleep for %v and check again.", aws.StringValue(bucket), timeBetweenRetries)

remote/remote_state_gcs.go
120:            terragruntOptions.Logger.Printf("Backend type has changed from gcs to %s", existingBackend.Type)
248:            terragruntOptions.Logger.Printf("Remote state GCS bucket %s does not exist. Attempting to create it", config.remoteStateConfigGCS.Bucket)
293:            terragruntOptions.Logger.Printf("WARNING: Versioning is not enabled for the remote state GCS bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
320:            terragruntOptions.Logger.Printf("No labels specified for bucket %s.", config.remoteStateConfigGCS.Bucket)
324:    terragruntOptions.Logger.Printf("Adding labels to GCS bucket with %s", config.GCSBucketLabels)
347:    terragruntOptions.Logger.Printf("Creating GCS bucket %s in project %s", config.remoteStateConfigGCS.Bucket, config.Project)
360:            terragruntOptions.Logger.Printf("Creating GCS bucket in location %s.", config.Location)
365:            terragruntOptions.Logger.Printf("Versioning is disabled for the remote state GCS bucket %s using 'skip_bucket_versioning' config.", config.remoteStateConfigGCS.Bucket)
367:            terragruntOptions.Logger.Printf("Enabling versioning on GCS bucket %s", config.remoteStateConfigGCS.Bucket)
372:            terragruntOptions.Logger.Printf("Enabling uniform bucket-level access on GCS bucket %s", config.remoteStateConfigGCS.Bucket)
383:    terragruntOptions.Logger.Printf("Waiting for bucket %s to be created", config.Bucket)
386:                    terragruntOptions.Logger.Printf("GCS bucket %s created.", config.Bucket)
389:                    terragruntOptions.Logger.Printf("GCS bucket %s has not been created yet. Sleeping for %s and will check again.", config.Bucket, SLEEP_BETWEEN_RETRIES_WAITING_FOR_GCS_BUCKET)
```

amnk commented:
@yorinasub17 Thanks for taking care of this!

@brikis98 I have pushed a change that changes all `Printf()` to `Infof()`. While this does not change any functionality, it does expose all of those for review in Github. After those are reviewed, we can come back to explicit `Infof()`:
```
config/dependency.go
472:    terragruntOptions.Logger.Infof("Detected module %s is already init-ed. Retrieving outputs directly from working directory.", targetConfig)

shell/run_shell_cmd.go
60:     terragruntOptions.Logger.Infof("Running command: %s %s", command, strings.Join(args, " "))

shell/prompt.go
28:             terragruntOptions.Logger.Infof("The non-interactive flag is set to true, so assuming 'yes' for all prompts")

cli/debug.go
24:     terragruntOptions.Logger.Infof(

cli/cli_app.go
357:            terragruntOptions.Logger.Infof(
925:    terragruntOptions.Logger.Infof("%s", stack.String())

remote/remote_state.go
110:            terragruntOptions.Logger.Infof("Backend type has changed from %s to %s", existingBackend.Type, remoteState.Backend)
```

brikis98 commented on cli/cli_app.go:
```diff
@@ -534,18 +534,18 @@ func processHooks(hooks []config.Hook, terragruntOptions *options.TerragruntOpti
 	errorsOccurred := []error{}
 
-	terragruntOptions.Logger.Printf("Detected %d Hooks", len(hooks))
+	terragruntOptions.Logger.Infof("Detected %d Hooks", len(hooks))
```
Debugf

brikis98 commented on cli/cli_app.go:
```diff
@@ -534,18 +534,18 @@ func processHooks(hooks []config.Hook, terragruntOptions *options.TerragruntOpti
 
 			if possibleError != nil {
-				terragruntOptions.Logger.Printf("Error running hook %s with message: %s", curHook.Name, possibleError.Error())
+				terragruntOptions.Logger.Infof("Error running hook %s with message: %s", curHook.Name, possibleError.Error())
```
Errorf

brikis98 commented on cli/cli_app.go:
```diff
@@ -674,7 +674,7 @@ func runActionWithHooks(description string, terragruntOptions *options.Terragrun
 		actionErrors = action()
 	} else {
-		terragruntOptions.Logger.Printf("Errors encountered running before_hooks. Not running '%s'.", description)
+		terragruntOptions.Logger.Infof("Errors encountered running before_hooks. Not running '%s'.", description)
```
Errorf

brikis98 commented on cli/download_source.go:
```diff
@@ -33,14 +33,14 @@ func downloadTerraformSource(source string, terragruntOptions *options.Terragrun
 	}
 
-	terragruntOptions.Logger.Printf("Copying files from %s into %s", terragruntOptions.WorkingDir, terraformSource.WorkingDir)
+	terragruntOptions.Logger.Infof("Copying files from %s into %s", terragruntOptions.WorkingDir, terraformSource.WorkingDir)
```
Debugf

brikis98 commented on cli/download_source.go:
```diff
@@ -33,14 +33,14 @@ func downloadTerraformSource(source string, terragruntOptions *options.Terragrun
 	updatedTerragruntOptions := terragruntOptions.Clone(terragruntOptions.TerragruntConfigPath)
 
-	terragruntOptions.Logger.Printf("Setting working directory to %s", terraformSource.WorkingDir)
+	terragruntOptions.Logger.Infof("Setting working directory to %s", terraformSource.WorkingDir)
```
Debugf

brikis98 commented on cli/download_source.go:
```diff
@@ -49,7 +49,7 @@ func downloadTerraformSource(source string, terragruntOptions *options.Terragrun
 func downloadTerraformSourceIfNecessary(terraformSource *tfsource.TerraformSource, terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
 	if terragruntOptions.SourceUpdate {
-		terragruntOptions.Logger.Printf("The --%s flag is set, so deleting the temporary folder %s before downloading source.", OPT_TERRAGRUNT_SOURCE_UPDATE, terraformSource.DownloadDir)
+		terragruntOptions.Logger.Infof("The --%s flag is set, so deleting the temporary folder %s before downloading source.", OPT_TERRAGRUNT_SOURCE_UPDATE, terraformSource.DownloadDir)
```
Debugf

brikis98 commented on cli/download_source.go:
```diff
@@ -61,7 +61,7 @@ func downloadTerraformSourceIfNecessary(terraformSource *tfsource.TerraformSourc
 
 	if alreadyLatest {
-		terragruntOptions.Logger.Printf("Terraform files in %s are up to date. Will not download again.", terraformSource.WorkingDir)
+		terragruntOptions.Logger.Infof("Terraform files in %s are up to date. Will not download again.", terraformSource.WorkingDir)
```
Debugf

brikis98 commented on cli/download_source.go:
```diff
@@ -104,7 +104,7 @@ func alreadyHaveLatestCode(terraformSource *tfsource.TerraformSource, terragrunt
 
 	if len(tfFiles) == 0 {
-		terragruntOptions.Logger.Printf("Working dir %s exists but contains no Terraform files, so assuming code needs to be downloaded again.", terraformSource.WorkingDir)
+		terragruntOptions.Logger.Infof("Working dir %s exists but contains no Terraform files, so assuming code needs to be downloaded again.", terraformSource.WorkingDir)
```
Debugf

brikis98 commented on cli/download_source.go:
```diff
@@ -146,7 +146,7 @@ var copyFiles = func(client *getter.Client) error {
 // Download the code from the Canonical Source URL into the Download Folder using the go-getter library
 func downloadSource(terraformSource *tfsource.TerraformSource, terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
-	terragruntOptions.Logger.Printf("Downloading Terraform configurations from %s into %s", terraformSource.CanonicalSourceURL, terraformSource.DownloadDir)
+	terragruntOptions.Logger.Infof("Downloading Terraform configurations from %s into %s", terraformSource.CanonicalSourceURL, terraformSource.DownloadDir)
```
Debugf

brikis98 commented on cli/hclfmt.go:
```diff
@@ -66,24 +66,24 @@ func runHCLFmt(terragruntOptions *options.TerragruntOptions) error {
 	info, err := os.Stat(tgHclFile)
 	if err != nil {
-		terragruntOptions.Logger.Printf("Error retrieving file info of %s", tgHclFile)
+		terragruntOptions.Logger.Infof("Error retrieving file info of %s", tgHclFile)
```
Errorf

brikis98 commented on cli/hclfmt.go:
```diff
@@ -66,24 +66,24 @@ func runHCLFmt(terragruntOptions *options.TerragruntOptions) error {
 	contentsStr, err := util.ReadFileAsString(tgHclFile)
 	if err != nil {
-		terragruntOptions.Logger.Printf("Error reading %s", tgHclFile)
+		terragruntOptions.Logger.Infof("Error reading %s", tgHclFile)
```
Errorf

brikis98 commented on cli/hclfmt.go:
```diff
@@ -66,24 +66,24 @@ func runHCLFmt(terragruntOptions *options.TerragruntOptions) error {
 	err = checkErrors(contents, tgHclFile)
 	if err != nil {
-		terragruntOptions.Logger.Printf("Error parsing %s", tgHclFile)
+		terragruntOptions.Logger.Infof("Error parsing %s", tgHclFile)
```
Errorf

brikis98 commented on cli/version_check.go:
```diff
@@ -47,7 +47,7 @@ func PopulateTerraformVersion(terragruntOptions *options.TerragruntOptions) erro
 
 	terragruntOptions.TerraformVersion = terraformVersion
-	terragruntOptions.Logger.Printf("Terraform version: %s", terraformVersion)
+	terragruntOptions.Logger.Infof("Terraform version: %s", terraformVersion)
```
Debugf

brikis98 commented on configstack/running_module.go:
```diff
@@ -249,7 +249,7 @@ func (module *runningModule) moduleFinished(moduleErr error) {
 		module.Module.TerragruntOptions.Logger.Debugf("Module %s has finished successfully!", module.Module.Path)
 	} else {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished with an error: %v", module.Module.Path, moduleErr)
+		module.Module.TerragruntOptions.Logger.Infof("Module %s has finished with an error: %v", module.Module.Path, moduleErr)
```
Errorf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -44,7 +44,7 @@ func CreateLockTableIfNecessary(tableName string, tags map[string]string, client
 
 	if !tableExists {
-		terragruntOptions.Logger.Printf("Lock table %s does not exist in DynamoDB. Will need to create it just this first time.", tableName)
+		terragruntOptions.Logger.Infof("Lock table %s does not exist in DynamoDB. Will need to create it just this first time.", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -81,7 +81,7 @@ func CreateLockTable(tableName string, tags map[string]string, client *dynamodb.
 	defer tableCreateDeleteSemaphore.Release()
 
-	terragruntOptions.Logger.Printf("Creating table %s in DynamoDB", tableName)
+	terragruntOptions.Logger.Infof("Creating table %s in DynamoDB", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -100,7 +100,7 @@ func CreateLockTable(tableName string, tags map[string]string, client *dynamodb.
 	if err != nil {
 		if isTableAlreadyBeingCreatedOrUpdatedError(err) {
-			terragruntOptions.Logger.Printf("Looks like someone created table %s at the same time. Will wait for it to be in active state.", tableName)
+			terragruntOptions.Logger.Infof("Looks like someone created table %s at the same time. Will wait for it to be in active state.", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -128,12 +128,12 @@ func CreateLockTable(tableName string, tags map[string]string, client *dynamodb.
 
 	if tags == nil || len(tags) == 0 {
-		terragruntOptions.Logger.Printf("No tags for lock table given.")
+		terragruntOptions.Logger.Infof("No tags for lock table given.")
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -128,12 +128,12 @@ func CreateLockTable(tableName string, tags map[string]string, client *dynamodb.
 
 	// we were able to create the table successfully, now add tags
-	terragruntOptions.Logger.Printf("Adding tags to lock table: %s", tags)
+	terragruntOptions.Logger.Infof("Adding tags to lock table: %s", tags)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -183,12 +183,12 @@ func waitForTableToBeActiveWithRandomSleep(tableName string, client *dynamodb.Dy
 
 		if tableReady {
-			terragruntOptions.Logger.Printf("Success! Table %s is now in active state.", tableName)
+			terragruntOptions.Logger.Infof("Success! Table %s is now in active state.", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -183,12 +183,12 @@ func waitForTableToBeActiveWithRandomSleep(tableName string, client *dynamodb.Dy
 
 		sleepBetweenRetries := util.GetRandomTime(sleepBetweenRetriesMin, sleepBetweenRetriesMax)
-		terragruntOptions.Logger.Printf("Table %s is not yet in active state. Will check again after %s.", tableName, sleepBetweenRetries)
+		terragruntOptions.Logger.Infof("Table %s is not yet in active state. Will check again after %s.", tableName, sleepBetweenRetries)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -203,14 +203,14 @@ func UpdateLockTableSetSSEncryptionOnIfNecessary(tableName string, client *dynam
 
 	if tableSSEncrypted {
-		terragruntOptions.Logger.Printf("Table %s already has encryption enabled", tableName)
+		terragruntOptions.Logger.Infof("Table %s already has encryption enabled", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -203,14 +203,14 @@ func UpdateLockTableSetSSEncryptionOnIfNecessary(tableName string, client *dynam
 	defer tableCreateDeleteSemaphore.Release()
 
-	terragruntOptions.Logger.Printf("Enabling server-side encryption on table %s in AWS DynamoDB", tableName)
+	terragruntOptions.Logger.Infof("Enabling server-side encryption on table %s in AWS DynamoDB", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -222,7 +222,7 @@ func UpdateLockTableSetSSEncryptionOnIfNecessary(tableName string, client *dynam
 	if _, err := client.UpdateTable(input); err != nil {
 		if isTableAlreadyBeingCreatedOrUpdatedError(err) {
-			terragruntOptions.Logger.Printf("Looks like someone is already updating table %s at the same time. Will wait for that update to complete.", tableName)
+			terragruntOptions.Logger.Infof("Looks like someone is already updating table %s at the same time. Will wait for that update to complete.", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -240,7 +240,7 @@ func waitForEncryptionToBeEnabled(tableName string, client *dynamodb.DynamoDB, t
 	sleepBetweenRetries := 20 * time.Second
 
-	terragruntOptions.Logger.Printf("Waiting for encryption to be enabled on table %s", tableName)
+	terragruntOptions.Logger.Infof("Waiting for encryption to be enabled on table %s", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -249,11 +249,11 @@ func waitForEncryptionToBeEnabled(tableName string, client *dynamodb.DynamoDB, t
 
 		if tableSSEncrypted {
-			terragruntOptions.Logger.Printf("Encryption is now enabled for table %s!", tableName)
+			terragruntOptions.Logger.Infof("Encryption is now enabled for table %s!", tableName)
```
Debugf

brikis98 commented on dynamodb/dynamo_lock_table.go:
```diff
@@ -249,11 +249,11 @@ func waitForEncryptionToBeEnabled(tableName string, client *dynamodb.DynamoDB, t
 		}
 
-		terragruntOptions.Logger.Printf("Encryption is still not enabled for table %s. Will sleep for %v and try again.", tableName, sleepBetweenRetries)
+		terragruntOptions.Logger.Infof("Encryption is still not enabled for table %s. Will sleep for %v and try again.", tableName, sleepBetweenRetries)
```
Debugf

brikis98 commented on options/options.go:
```diff
@@ -227,7 +227,7 @@ func NewTerragruntOptionsForTest(terragruntConfigPath string) (*TerragruntOption
 	if err != nil {
 		logger := util.CreateLogEntry("", DEFAULT_LOG_LEVEL)
-		logger.Printf("error: %v\n", errors.WithStackTrace(err))
+		logger.Infof("error: %v\n", errors.WithStackTrace(err))
```
Errorf

brikis98 commented on remote/remote_state.go:
```diff
@@ -64,7 +64,7 @@ func (remoteState *RemoteState) Validate() error {
 // using S3 or GCS for remote state storage, this may create the bucket if it doesn't exist already.
 func (remoteState *RemoteState) Initialize(terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Initializing remote state for the %s backend", remoteState.Backend)
+	terragruntOptions.Logger.Infof("Initializing remote state for the %s backend", remoteState.Backend)
```
Debugf

brikis98 commented on remote/remote_state.go:
```diff
@@ -116,7 +116,7 @@ func (remoteState *RemoteState) differsFrom(existingBackend *TerraformBackend, t
 	}
 
-	terragruntOptions.Logger.Printf("Backend %s has not changed.", existingBackend.Type)
+	terragruntOptions.Logger.Infof("Backend %s has not changed.", existingBackend.Type)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -117,7 +117,7 @@ func gcsConfigValuesEqual(config map[string]interface{}, existingBackend *Terraf
 
 	if existingBackend.Type != "gcs" {
-		terragruntOptions.Logger.Printf("Backend type has changed from gcs to %s", existingBackend.Type)
+		terragruntOptions.Logger.Infof("Backend type has changed from gcs to %s", existingBackend.Type)
```
Debugf

brikis98 replied:
I think this one still needs to be tweaked

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -245,7 +245,7 @@ func validateGCSConfig(extendedConfig *ExtendedRemoteStateConfigGCS, terragruntO
 func createGCSBucketIfNecessary(gcsClient *storage.Client, config *ExtendedRemoteStateConfigGCS, terragruntOptions *options.TerragruntOptions) error {
 	if !DoesGCSBucketExist(gcsClient, &config.remoteStateConfigGCS) {
-		terragruntOptions.Logger.Printf("Remote state GCS bucket %s does not exist. Attempting to create it", config.remoteStateConfigGCS.Bucket)
+		terragruntOptions.Logger.Infof("Remote state GCS bucket %s does not exist. Attempting to create it", config.remoteStateConfigGCS.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -290,7 +290,7 @@ func checkIfGCSVersioningEnabled(gcsClient *storage.Client, config *RemoteStateC
 
 	if attrs.VersioningEnabled == false {
-		terragruntOptions.Logger.Printf("WARNING: Versioning is not enabled for the remote state GCS bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
+		terragruntOptions.Logger.Infof("WARNING: Versioning is not enabled for the remote state GCS bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
```
Warnf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -317,11 +317,11 @@ func CreateGCSBucketWithVersioning(gcsClient *storage.Client, config *ExtendedRe
 func AddLabelsToGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfigGCS, terragruntOptions *options.TerragruntOptions) error {
 	if config.GCSBucketLabels == nil || len(config.GCSBucketLabels) == 0 {
-		terragruntOptions.Logger.Printf("No labels specified for bucket %s.", config.remoteStateConfigGCS.Bucket)
+		terragruntOptions.Logger.Infof("No labels specified for bucket %s.", config.remoteStateConfigGCS.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -317,11 +317,11 @@ func CreateGCSBucketWithVersioning(gcsClient *storage.Client, config *ExtendedRe
 	}
 
-	terragruntOptions.Logger.Printf("Adding labels to GCS bucket with %s", config.GCSBucketLabels)
+	terragruntOptions.Logger.Infof("Adding labels to GCS bucket with %s", config.GCSBucketLabels)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -344,7 +344,7 @@ func AddLabelsToGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteState
 // Create the GCS bucket specified in the given config
 func CreateGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfigGCS, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Creating GCS bucket %s in project %s", config.remoteStateConfigGCS.Bucket, config.Project)
+	terragruntOptions.Logger.Infof("Creating GCS bucket %s in project %s", config.remoteStateConfigGCS.Bucket, config.Project)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -357,19 +357,19 @@ func CreateGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfi
 
 	if config.Location != "" {
-		terragruntOptions.Logger.Printf("Creating GCS bucket in location %s.", config.Location)
+		terragruntOptions.Logger.Infof("Creating GCS bucket in location %s.", config.Location)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -357,19 +357,19 @@ func CreateGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfi
 
 	if config.SkipBucketVersioning {
-		terragruntOptions.Logger.Printf("Versioning is disabled for the remote state GCS bucket %s using 'skip_bucket_versioning' config.", config.remoteStateConfigGCS.Bucket)
+		terragruntOptions.Logger.Infof("Versioning is disabled for the remote state GCS bucket %s using 'skip_bucket_versioning' config.", config.remoteStateConfigGCS.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -357,19 +357,19 @@ func CreateGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfi
+		terragruntOptions.Logger.Infof("Versioning is disabled for the remote state GCS bucket %s using 'skip_bucket_versioning' config.", config.remoteStateConfigGCS.Bucket)
 	} else {
-		terragruntOptions.Logger.Printf("Enabling versioning on GCS bucket %s", config.remoteStateConfigGCS.Bucket)
+		terragruntOptions.Logger.Infof("Enabling versioning on GCS bucket %s", config.remoteStateConfigGCS.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -357,19 +357,19 @@ func CreateGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfi
 
 	if config.EnableBucketPolicyOnly {
-		terragruntOptions.Logger.Printf("Enabling uniform bucket-level access on GCS bucket %s", config.remoteStateConfigGCS.Bucket)
+		terragruntOptions.Logger.Infof("Enabling uniform bucket-level access on GCS bucket %s", config.remoteStateConfigGCS.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -380,13 +380,13 @@ func CreateGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfi
 // about that GCS bucket has propagated everywhere.
 func WaitUntilGCSBucketExists(gcsClient *storage.Client, config *RemoteStateConfigGCS, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Waiting for bucket %s to be created", config.Bucket)
+	terragruntOptions.Logger.Infof("Waiting for bucket %s to be created", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -380,13 +380,13 @@ func CreateGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfi
 	for retries := 0; retries < MAX_RETRIES_WAITING_FOR_GCS_BUCKET; retries++ {
 		if DoesGCSBucketExist(gcsClient, config) {
-			terragruntOptions.Logger.Printf("GCS bucket %s created.", config.Bucket)
+			terragruntOptions.Logger.Infof("GCS bucket %s created.", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -380,13 +380,13 @@ func CreateGCSBucket(gcsClient *storage.Client, config *ExtendedRemoteStateConfi
 			return nil
 		} else if retries < MAX_RETRIES_WAITING_FOR_GCS_BUCKET-1 {
-			terragruntOptions.Logger.Printf("GCS bucket %s has not been created yet. Sleeping for %s and will check again.", config.Bucket, SLEEP_BETWEEN_RETRIES_WAITING_FOR_GCS_BUCKET)
+			terragruntOptions.Logger.Infof("GCS bucket %s has not been created yet. Sleeping for %s and will check again.", config.Bucket, SLEEP_BETWEEN_RETRIES_WAITING_FOR_GCS_BUCKET)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -125,7 +125,7 @@ func (s3Initializer S3Initializer) NeedsInitialization(remoteState *RemoteState,
 	// attribute is being used.
 	if util.KindOf(remoteState.Config["lock_table"]) == reflect.String && remoteState.Config["lock_table"] != "" {
-		terragruntOptions.Logger.Printf("%s\n", lockTableDeprecationMessage)
+		terragruntOptions.Logger.Infof("%s\n", lockTableDeprecationMessage)
```
Warnf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -176,7 +176,7 @@ func configValuesEqual(config map[string]interface{}, existingBackend *Terraform
 
 	if existingBackend.Type != "s3" {
-		terragruntOptions.Logger.Printf("Backend type has changed from s3 to %s", existingBackend.Type)
+		terragruntOptions.Logger.Infof("Backend type has changed from s3 to %s", existingBackend.Type)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -193,7 +193,7 @@ func configValuesEqual(config map[string]interface{}, existingBackend *Terraform
 			existingBackend.Config["encrypt"] = value
 		} else {
-			terragruntOptions.Logger.Printf("Remote state configuration encrypt contains invalid value %v, should be boolean.", existingBackend.Config["encrypt"])
+			terragruntOptions.Logger.Infof("Remote state configuration encrypt contains invalid value %v, should be boolean.", existingBackend.Config["encrypt"])
```
Warnf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -240,7 +240,7 @@ func (s3Initializer S3Initializer) Initialize(remoteState *RemoteState, terragru
 	// during initialization.
 	if s3Config.LockTable != "" {
-		terragruntOptions.Logger.Printf("%s\n", lockTableDeprecationMessage)
+		terragruntOptions.Logger.Infof("%s\n", lockTableDeprecationMessage)
```
Warnf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -334,7 +334,7 @@ func validateS3Config(extendedConfig *ExtendedRemoteStateConfigS3, terragruntOpt
 
 	if !config.Encrypt {
-		terragruntOptions.Logger.Printf("WARNING: encryption is not enabled on the S3 remote state bucket %s. Terraform state files may contain secrets, so we STRONGLY recommend enabling encryption!", config.Bucket)
+		terragruntOptions.Logger.Infof("WARNING: encryption is not enabled on the S3 remote state bucket %s. Terraform state files may contain secrets, so we STRONGLY recommend enabling encryption!", config.Bucket)
```
Warnf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -380,7 +380,7 @@ func checkIfVersioningEnabled(s3Client *s3.S3, config *RemoteStateConfigS3, terr
 	// check the AWS SDK for updates to see if we can remove "out == nil ||".
 	if out == nil || out.Status == nil || *out.Status != s3.BucketVersioningStatusEnabled {
-		terragruntOptions.Logger.Printf("WARNING: Versioning is not enabled for the remote state S3 bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
+		terragruntOptions.Logger.Infof("WARNING: Versioning is not enabled for the remote state S3 bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
```
Warnf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -392,7 +392,7 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 	if err != nil {
 		if isBucketAlreadyOwnedByYouError(err) {
-			terragruntOptions.Logger.Printf("Looks like you're already creating bucket %s at the same time. Will not attempt to create it again.", config.remoteStateConfigS3.Bucket)
+			terragruntOptions.Logger.Infof("Looks like you're already creating bucket %s at the same time. Will not attempt to create it again.", config.remoteStateConfigS3.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -404,13 +404,13 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 	if config.SkipBucketRootAccess {
-		terragruntOptions.Logger.Printf("Root access is disabled for the remote state S3 bucket %s using 'skip_bucket_root_access' config.", config.remoteStateConfigS3.Bucket)
+		terragruntOptions.Logger.Infof("Root access is disabled for the remote state S3 bucket %s using 'skip_bucket_root_access' config.", config.remoteStateConfigS3.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -404,13 +404,13 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 	if config.SkipBucketEnforcedTLS {
-		terragruntOptions.Logger.Printf("TLS enforcement is disabled for the remote state S3 bucket %s using 'skip_bucket_enforced_tls' config.", config.remoteStateConfigS3.Bucket)
+		terragruntOptions.Logger.Infof("TLS enforcement is disabled for the remote state S3 bucket %s using 'skip_bucket_enforced_tls' config.", config.remoteStateConfigS3.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -424,34 +424,34 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 	if config.SkipBucketVersioning {
-		terragruntOptions.Logger.Printf("Versioning is disabled for the remote state S3 bucket %s using 'skip_bucket_versioning' config.", config.remoteStateConfigS3.Bucket)
+		terragruntOptions.Logger.Infof("Versioning is disabled for the remote state S3 bucket %s using 'skip_bucket_versioning' config.", config.remoteStateConfigS3.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -424,34 +424,34 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 	if config.SkipBucketSSEncryption {
-		terragruntOptions.Logger.Printf("Server-Side Encryption is disabled for the remote state AWS S3 bucket %s using 'skip_bucket_ssencryption' config.", config.remoteStateConfigS3.Bucket)
+		terragruntOptions.Logger.Infof("Server-Side Encryption is disabled for the remote state AWS S3 bucket %s using 'skip_bucket_ssencryption' config.", config.remoteStateConfigS3.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -424,34 +424,34 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 	if config.SkipBucketAccessLogging {
-		terragruntOptions.Logger.Printf("___WARNING___: The terragrunt configuration option 'skip_bucket_accesslogging' is now deprecated. Access logging for the state bucket %s is disabled by default. To enable access logging for bucket %s, please provide property `accesslogging_bucket_name` in the terragrunt config file. For more details, please refer to the Terragrunt documentation.", config.remoteS ...
+		terragruntOptions.Logger.Infof("___WARNING___: The terragrunt configuration option 'skip_bucket_accesslogging' is now deprecated. Access logging for the state bucket %s is disabled by default. To enable access logging for bucket %s, please provide property `accesslogging_bucket_name` in the terragrunt config file. For more details, please refer to the Terragrunt documentation.", config.remoteSt ...
```
Warnf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -424,34 +424,34 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 	if config.AccessLoggingBucketName != "" {
-		terragruntOptions.Logger.Printf("Enabling bucket-wide Access Logging on AWS S3 bucket %s - using as TargetBucket %s", config.remoteStateConfigS3.Bucket, config.AccessLoggingBucketName)
+		terragruntOptions.Logger.Infof("Enabling bucket-wide Access Logging on AWS S3 bucket %s - using as TargetBucket %s", config.remoteStateConfigS3.Bucket, config.AccessLoggingBucketName)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -424,34 +424,34 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 		if err := CreateLogsS3BucketIfNecessary(s3Client, aws.String(config.AccessLoggingBucketName), terragruntOptions); err != nil {
-			terragruntOptions.Logger.Printf("Error: Could not create logs bucket %s for AWS S3 bucket %s", config.AccessLoggingBucketName, config.remoteStateConfigS3.Bucket)
+			terragruntOptions.Logger.Infof("Error: Could not create logs bucket %s for AWS S3 bucket %s", config.AccessLoggingBucketName, config.remoteStateConfigS3.Bucket)
```
Errorf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -424,34 +424,34 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 		}
 	} else {
-		terragruntOptions.Logger.Printf("Access Logging is disabled for the remote state AWS S3 bucket %s", config.remoteStateConfigS3.Bucket)
+		terragruntOptions.Logger.Infof("Access Logging is disabled for the remote state AWS S3 bucket %s", config.remoteStateConfigS3.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -475,14 +475,14 @@ func CreateLogsS3BucketIfNecessary(s3Client *s3.S3, logsBucketName *string, terr
 
 	if config.S3BucketTags == nil || len(config.S3BucketTags) == 0 {
-		terragruntOptions.Logger.Printf("No tags specified for bucket %s.", config.remoteStateConfigS3.Bucket)
+		terragruntOptions.Logger.Infof("No tags specified for bucket %s.", config.remoteStateConfigS3.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -475,14 +475,14 @@ func CreateLogsS3BucketIfNecessary(s3Client *s3.S3, logsBucketName *string, terr
 	var tagsConverted = convertTags(config.S3BucketTags)
 
-	terragruntOptions.Logger.Printf("Tagging S3 bucket with %s", config.S3BucketTags)
+	terragruntOptions.Logger.Infof("Tagging S3 bucket with %s", config.S3BucketTags)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -494,7 +494,7 @@ func TagS3Bucket(s3Client *s3.S3, config *ExtendedRemoteStateConfigS3, terragrun
 	}
 
-	terragruntOptions.Logger.Printf("Tagged S3 bucket with %s", config.S3BucketTags)
+	terragruntOptions.Logger.Infof("Tagged S3 bucket with %s", config.S3BucketTags)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -516,13 +516,13 @@ func convertTags(tags map[string]string) []*s3.Tag {
 // about that S3 bucket has propagated everywhere
 func WaitUntilS3BucketExists(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Waiting for bucket %s to be created", config.Bucket)
+	terragruntOptions.Logger.Infof("Waiting for bucket %s to be created", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -516,13 +516,13 @@ func convertTags(tags map[string]string) []*s3.Tag {
 	for retries := 0; retries < MAX_RETRIES_WAITING_FOR_S3_BUCKET; retries++ {
 		if DoesS3BucketExist(s3Client, aws.String(config.Bucket)) {
-			terragruntOptions.Logger.Printf("S3 bucket %s created.", config.Bucket)
+			terragruntOptions.Logger.Infof("S3 bucket %s created.", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -516,13 +516,13 @@ func convertTags(tags map[string]string) []*s3.Tag {
 			return nil
 		} else if retries < MAX_RETRIES_WAITING_FOR_S3_BUCKET-1 {
-			terragruntOptions.Logger.Printf("S3 bucket %s has not been created yet. Sleeping for %s and will check again.", config.Bucket, SLEEP_BETWEEN_RETRIES_WAITING_FOR_S3_BUCKET)
+			terragruntOptions.Logger.Infof("S3 bucket %s has not been created yet. Sleeping for %s and will check again.", config.Bucket, SLEEP_BETWEEN_RETRIES_WAITING_FOR_S3_BUCKET)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -532,12 +532,12 @@ func WaitUntilS3BucketExists(s3Client *s3.S3, config *RemoteStateConfigS3, terra
 // Create the S3 bucket specified in the given config
 func CreateS3Bucket(s3Client *s3.S3, bucket *string, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Creating S3 bucket %s", aws.StringValue(bucket))
+	terragruntOptions.Logger.Infof("Creating S3 bucket %s", aws.StringValue(bucket))
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -532,12 +532,12 @@ func WaitUntilS3BucketExists(s3Client *s3.S3, config *RemoteStateConfigS3, terra
 		return errors.WithStackTrace(err)
 	}
-	terragruntOptions.Logger.Printf("Created S3 bucket %s", aws.StringValue(bucket))
+	terragruntOptions.Logger.Infof("Created S3 bucket %s", aws.StringValue(bucket))
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -550,7 +550,7 @@ func isBucketAlreadyOwnedByYouError(err error) bool {
 // Add a policy to allow root access to the bucket
 func EnableRootAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Enabling root access to S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabling root access to S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -590,13 +590,13 @@ func EnableRootAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, te
 	}
 
-	terragruntOptions.Logger.Printf("Enabled root access to bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabled root access to bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -590,13 +590,13 @@ func EnableRootAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, te
 // Add a policy to enforce TLS based access to the bucket
 func EnableEnforcedTLSAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Enabling enforced TLS access for S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabling enforced TLS access for S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -632,13 +632,13 @@ func EnableEnforcedTLSAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfi
 	}
 
-	terragruntOptions.Logger.Printf("Enabled enforced TLS access for bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabled enforced TLS access for bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -632,13 +632,13 @@ func EnableEnforcedTLSAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfi
 // Enable versioning for the S3 bucket specified in the given config
 func EnableVersioningForS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Enabling versioning on S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabling versioning on S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -649,13 +649,13 @@ func EnableVersioningForS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, t
 	}
 
-	terragruntOptions.Logger.Printf("Enabled versioning on S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabled versioning on S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -649,13 +649,13 @@ func EnableVersioningForS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, t
 // Enable bucket-wide Server-Side Encryption for the AWS S3 bucket specified in the given config
 func EnableSSEForS3BucketWide(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Enabling bucket-wide SSE on AWS S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabling bucket-wide SSE on AWS S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -668,7 +668,7 @@ func EnableSSEForS3BucketWide(s3Client *s3.S3, config *RemoteStateConfigS3, terr
 	}
 
-	terragruntOptions.Logger.Printf("Enabled bucket-wide SSE on AWS S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabled bucket-wide SSE on AWS S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -679,7 +679,7 @@ func EnableAccessLoggingForS3BucketWide(s3Client *s3.S3, config *RemoteStateConf
 
 	targetPrefix := "TFStateLogs/"
-	terragruntOptions.Logger.Printf("Putting bucket logging on S3 bucket %s with TargetBucket %s and TargetPrefix %s", config.Bucket, logsBucket, targetPrefix)
+	terragruntOptions.Logger.Infof("Putting bucket logging on S3 bucket %s with TargetBucket %s and TargetPrefix %s", config.Bucket, logsBucket, targetPrefix)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -695,15 +695,15 @@ func EnableAccessLoggingForS3BucketWide(s3Client *s3.S3, config *RemoteStateConf
 	}
 
-	terragruntOptions.Logger.Printf("Enabled bucket-wide Access Logging on AWS S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Enabled bucket-wide Access Logging on AWS S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -695,15 +695,15 @@ func EnableAccessLoggingForS3BucketWide(s3Client *s3.S3, config *RemoteStateConf
 // https://docs.aws.amazon.com/AmazonS3/latest/dev/access-control-block-public-access.html for more information.
 func EnablePublicAccessBlockingForS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Blocking all public access to S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Blocking all public access to S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -720,15 +720,15 @@ func EnablePublicAccessBlockingForS3Bucket(s3Client *s3.S3, config *RemoteStateC
 	}
 
-	terragruntOptions.Logger.Printf("Blocked all public access to S3 bucket %s", config.Bucket)
+	terragruntOptions.Logger.Infof("Blocked all public access to S3 bucket %s", config.Bucket)
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -720,15 +720,15 @@ func EnablePublicAccessBlockingForS3Bucket(s3Client *s3.S3, config *RemoteStateC
 // https://docs.aws.amazon.com/AmazonS3/latest/dev/enable-logging-programming.html
 func configureBucketAccessLoggingAcl(s3Client *s3.S3, bucket *string, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Granting WRITE and READ_ACP permissions to S3 Log Delivery (%s) for bucket %s. This is required for access logging.", s3LogDeliveryGranteeUri, aws.StringValue(bucket))
+	terragruntOptions.Logger.Infof("Granting WRITE and READ_ACP permissions to S3 Log Delivery (%s) for bucket %s. This is required for access logging.", s3LogDeliveryGranteeUri, aws.StringValue(bucket))
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -745,7 +745,7 @@ func configureBucketAccessLoggingAcl(s3Client *s3.S3, bucket *string, terragrunt
 
 func waitUntilBucketHasAccessLoggingAcl(s3Client *s3.S3, bucket *string, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Waiting for ACL bucket %s to have the updated ACL for access logging.", aws.StringValue(bucket))
+	terragruntOptions.Logger.Infof("Waiting for ACL bucket %s to have the updated ACL for access logging.", aws.StringValue(bucket))
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -771,11 +771,11 @@ func waitUntilBucketHasAccessLoggingAcl(s3Client *s3.S3, bucket *string, terragr
 
 		if hasReadAcp && hasWrite {
-			terragruntOptions.Logger.Printf("Bucket %s now has the proper ACL permissions for access logging!", aws.StringValue(bucket))
+			terragruntOptions.Logger.Infof("Bucket %s now has the proper ACL permissions for access logging!", aws.StringValue(bucket))
```
Debugf

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -771,11 +771,11 @@ func waitUntilBucketHasAccessLoggingAcl(s3Client *s3.S3, bucket *string, terragr
 		}
 
-		terragruntOptions.Logger.Printf("Bucket %s still does not have the ACL permissions for access logging. Will sleep for %v and check again.", aws.StringValue(bucket), timeBetweenRetries)
+		terragruntOptions.Logger.Infof("Bucket %s still does not have the ACL permissions for access logging. Will sleep for %v and check again.", aws.StringValue(bucket), timeBetweenRetries)
```
Debugf

brikis98 commented on shell/ptty_unix.go:
```diff
@@ -27,7 +27,7 @@ func runCommandWithPTTY(terragruntOptions *options.TerragruntOptions, cmd *exec.
 	defer func() {
 		if closeErr := pseudoTerminal.Close(); closeErr != nil {
-			terragruntOptions.Logger.Printf("Error closing pty: %s", closeErr)
+			terragruntOptions.Logger.Infof("Error closing pty: %s", closeErr)
```
Errorf

brikis98 commented on shell/ptty_unix.go:
```diff
@@ -45,7 +45,7 @@ func runCommandWithPTTY(terragruntOptions *options.TerragruntOptions, cmd *exec.
 		for range ch {
 			if inheritSizeErr := pty.InheritSize(os.Stdin, pseudoTerminal); inheritSizeErr != nil {
-				terragruntOptions.Logger.Printf("error resizing pty: %s", inheritSizeErr)
+				terragruntOptions.Logger.Infof("error resizing pty: %s", inheritSizeErr)
```
Errorf

brikis98 commented on shell/ptty_unix.go:
```diff
@@ -59,7 +59,7 @@ func runCommandWithPTTY(terragruntOptions *options.TerragruntOptions, cmd *exec.
 	defer func() {
 		if restoreErr := terminal.Restore(int(os.Stdin.Fd()), oldState); restoreErr != nil {
-			terragruntOptions.Logger.Printf("Error restoring terminal state: %s", restoreErr)
+			terragruntOptions.Logger.Infof("Error restoring terminal state: %s", restoreErr)
```
Errorf

brikis98 commented on shell/ptty_unix.go:
```diff
@@ -71,7 +71,7 @@ func runCommandWithPTTY(terragruntOptions *options.TerragruntOptions, cmd *exec.
 	go func() {
 		_, copyStdinErr := io.Copy(pseudoTerminal, os.Stdin)
-		terragruntOptions.Logger.Printf("Error forwarding stdin: %s", copyStdinErr)
+		terragruntOptions.Logger.Infof("Error forwarding stdin: %s", copyStdinErr)
```
Errorf

brikis98 commented on shell/ptty_windows.go:
```diff
@@ -26,12 +26,12 @@ func enableVirtualTerminalProcessing(options *options.TerragruntOptions, file *o
 	handle := windows.Handle(file.Fd())
 	if err := windows.GetConsoleMode(handle, &mode); err != nil {
-		options.Logger.Printf("failed to get console mode: %v\n", err)
+		options.Logger.Infof("failed to get console mode: %v\n", err)
```
Errorf

brikis98 commented on shell/ptty_windows.go:
```diff
@@ -26,12 +26,12 @@ func enableVirtualTerminalProcessing(options *options.TerragruntOptions, file *o
 
 	if err := windows.SetConsoleMode(handle, mode|windows.ENABLE_VIRTUAL_TERMINAL_PROCESSING); err != nil {
-		options.Logger.Printf("failed to set console mode: %v\n", err)
+		options.Logger.Infof("failed to set console mode: %v\n", err)
```
Errorf

brikis98 commented on shell/run_shell_cmd.go:
```diff
@@ -174,10 +174,10 @@ func NewSignalsForwarder(signals []os.Signal, c *exec.Cmd, logger *logrus.Entry,
 			select {
 			case s := <-signalChannel:
-				logger.Printf("Forward signal %v to terraform.", s)
+				logger.Infof("Forward signal %v to terraform.", s)
```
Debugf

brikis98 commented on shell/run_shell_cmd.go:
```diff
@@ -174,10 +174,10 @@ func NewSignalsForwarder(signals []os.Signal, c *exec.Cmd, logger *logrus.Entry,
 				err := c.Process.Signal(s)
 				if err != nil {
-					logger.Printf("Error forwarding signal: %v", err)
+					logger.Infof("Error forwarding signal: %v", err)
```
Errorf

brikis98 commented on util/retry.go:
```diff
@@ -11,7 +11,7 @@ import (
 // MaxRetriesExceeded error.
 func DoWithRetry(actionDescription string, maxRetries int, sleepBetweenRetries time.Duration, logger *logrus.Entry, action func() error) error {
```
This method will need to take in the log level to use for `actionDescription`. No way to know from inside the method otherwise if this should be Info or Debug or whatever else.

amnk replied:
@brikis98 this is used in two places only: in [remote/remote_state_gcs.go#L272](https://github.com/amnk/terragrunt/blob/amnk/change-default-log-level/remote/remote_state_gcs.go#L272) and [remote/remote_state_s3.go#L364](https://github.com/amnk/terragrunt/blob/amnk/change-default-log-level/remote/remote_state_s3.go#L364). 

I have changes the method to take logLevel as input, and configured both places as `Debugf`. Hope that's ok

brikis98 replied:
Looks great, thanks!

brikis98 reviewed (commented):
Thank you both!

brikis98 commented:
> Here's what I got from `rg "(Info)|(Print)"` and cutting out the extraneous ones. I probably could get more context for some of these where I only got the function with `(`, but am also a bit buried to do more than this.

Thanks Yori!

I believe @amnk changed the `Print` ones to `Info`, so just looking at the other `Info` ones you had:

> ```
> config/dependency.go
> 472:    terragruntOptions.Logger.Infof("Detected module %s is already init-ed. Retrieving outputs directly from working directory.", targetConfig)
> ```

Debugf

> ```
> shell/run_shell_cmd.go
> 60:     terragruntOptions.Logger.Infof("Running command: %s %s", command, strings.Join(args, " "))
> ```

Debugf

> ```
> shell/prompt.go
> 28:             terragruntOptions.Logger.Infof("The non-interactive flag is set to true, so assuming 'yes' for all prompts")
> ```

Leave as-is


> ```
> cli/debug.go
> 24:     terragruntOptions.Logger.Infof(
> ```

Leave as-is

> ```
> cli/cli_app.go
> 357:            terragruntOptions.Logger.Infof(
> ```

Leave as-is

> ```
> 925:    terragruntOptions.Logger.Infof("%s", stack.String())
> ```

Leave as-is

> ```
> remote/remote_state.go
> 110:            terragruntOptions.Logger.Infof("Backend type has changed from %s to %s", existingBackend.Type, remoteState.Backend)
> ```

Leave as-is

brikis98 commented on cli/cli_app.go:
```diff
@@ -534,18 +534,18 @@ func processHooks(hooks []config.Hook, terragruntOptions *options.TerragruntOpti
+			terragruntOptions.Logger.Infof("Executing hook: %s", curHook.Name)
 			actionToExecute := curHook.Execute[0]
 			actionParams := curHook.Execute[1:]
 			possibleError := shell.RunShellCommand(terragruntOptions, actionToExecute, actionParams...)
```
How is the log output handled by `RunShellCommand`? Are we hiding the logs of the hook scripts being run here?

amnk replied:
@brikis98 we are returning errors from [RunShellCommandWithOutput()](https://github.com/amnk/terragrunt/blob/amnk/change-default-log-level/shell/run_shell_cmd.go#L52) from what I can see

brikis98 replied:
But it's not just the final error, but also `stdout` / `stderr` for the shell command itself, right?

brikis98 replied:
Any update on whether this works correctly?

amnk replied:
This particular change only affects log line itself. I tested hook output, and it is shown under `Info`, so all is fine?

brikis98 replied:
As long as hook output is visible, that sounds good to me!

brikis98 commented on remote/remote_state_gcs.go:
```diff
@@ -290,7 +291,7 @@ func checkIfGCSVersioningEnabled(gcsClient *storage.Client, config *RemoteStateC
 
 	if attrs.VersioningEnabled == false {
-		terragruntOptions.Logger.Printf("WARNING: Versioning is not enabled for the remote state GCS bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
+		terragruntOptions.Logger.Warnf("WARNING: Versioning is not enabled for the remote state GCS bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
```
```suggestion
		terragruntOptions.Logger.Warnf("Versioning is not enabled for the remote state GCS bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
```

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -334,7 +335,7 @@ func validateS3Config(extendedConfig *ExtendedRemoteStateConfigS3, terragruntOpt
 
 	if !config.Encrypt {
-		terragruntOptions.Logger.Printf("WARNING: encryption is not enabled on the S3 remote state bucket %s. Terraform state files may contain secrets, so we STRONGLY recommend enabling encryption!", config.Bucket)
+		terragruntOptions.Logger.Warnf("WARNING: encryption is not enabled on the S3 remote state bucket %s. Terraform state files may contain secrets, so we STRONGLY recommend enabling encryption!", config.Bucket)
```
```suggestion
		terragruntOptions.Logger.Warnf("Encryption is not enabled on the S3 remote state bucket %s. Terraform state files may contain secrets, so we STRONGLY recommend enabling encryption!", config.Bucket)
```

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -380,7 +381,7 @@ func checkIfVersioningEnabled(s3Client *s3.S3, config *RemoteStateConfigS3, terr
 	// check the AWS SDK for updates to see if we can remove "out == nil ||".
 	if out == nil || out.Status == nil || *out.Status != s3.BucketVersioningStatusEnabled {
-		terragruntOptions.Logger.Printf("WARNING: Versioning is not enabled for the remote state S3 bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
+		terragruntOptions.Logger.Warnf("WARNING: Versioning is not enabled for the remote state S3 bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
```
```suggestion
		terragruntOptions.Logger.Warnf("Versioning is not enabled for the remote state S3 bucket %s. We recommend enabling versioning so that you can roll back to previous versions of your Terraform state in case of error.", config.Bucket)
```

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -424,34 +425,34 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 	if config.SkipBucketAccessLogging {
-		terragruntOptions.Logger.Printf("___WARNING___: The terragrunt configuration option 'skip_bucket_accesslogging' is now deprecated. Access logging for the state bucket %s is disabled by default. To enable access logging for bucket %s, please provide property `accesslogging_bucket_name` in the terragrunt config file. For more details, please refer to the Terragrunt documentation.", config.remoteS ...
+		terragruntOptions.Logger.Warnf("___WARNING___: The terragrunt configuration option 'skip_bucket_accesslogging' is now deprecated. Access logging for the state bucket %s is disabled by default. To enable access logging for bucket %s, please provide property `accesslogging_bucket_name` in the terragrunt config file. For more details, please refer to the Terragrunt documentation.", config.remoteSt ...
```
```suggestion
		terragruntOptions.Logger.Warnf("The terragrunt configuration option 'skip_bucket_accesslogging' is now deprecated. Access logging for the state bucket %s is disabled by default. To enable access logging for bucket %s, please provide property `accesslogging_bucket_name` in the terragrunt config file. For more details, please refer to the Terragrunt documentation.", config.remoteStateConfigS3.Bucket, config.remoteStateConfigS3.Bucket)
```

brikis98 commented on remote/remote_state_s3.go:
```diff
@@ -424,34 +425,34 @@ func CreateS3BucketWithVersioningSSEncryptionAndAccessLogging(s3Client *s3.S3, c
 
 		if err := CreateLogsS3BucketIfNecessary(s3Client, aws.String(config.AccessLoggingBucketName), terragruntOptions); err != nil {
-			terragruntOptions.Logger.Printf("Error: Could not create logs bucket %s for AWS S3 bucket %s", config.AccessLoggingBucketName, config.remoteStateConfigS3.Bucket)
+			terragruntOptions.Logger.Errorf("Error: Could not create logs bucket %s for AWS S3 bucket %s", config.AccessLoggingBucketName, config.remoteStateConfigS3.Bucket)
```
```suggestion
			terragruntOptions.Logger.Errorf("Could not create logs bucket %s for AWS S3 bucket %s", config.AccessLoggingBucketName, config.remoteStateConfigS3.Bucket)
```

brikis98 commented on options/options.go:
```diff
@@ -227,7 +227,7 @@ func NewTerragruntOptionsForTest(terragruntConfigPath string) (*TerragruntOption
 	if err != nil {
 		logger := util.CreateLogEntry("", DEFAULT_LOG_LEVEL)
-		logger.Printf("error: %v\n", errors.WithStackTrace(err))
+		logger.Errorf("error: %v\n", errors.WithStackTrace(err))
```
```suggestion
		logger.Errorf("%v\n", errors.WithStackTrace(err))
```

brikis98 reviewed (commented):
A few last tweaks

brikis98 commented:
Reviewing the related bugs:

> Related: #1529

I'm not sure if this PR addressed this or not. @amnk Could you try to repro and see if this issue is fixed with the logging updates in this PR?

> Related: #1530

I believe we are now using `Infof` to print out the stack, so this should be fixed.

> Related: #1531

I'm not sure if this issue is the same as #1530, or if it's about the need to hide logging prompts when `--non-interactive` is set. If it's the former, we already take care of it; if it's the latter, I guess we could make those prompts `Debugf` messages when `--non-interactive` is set, but write them directly to `stderr` otherwise?

> Related: #1532

This seems very close to #1529, where `run-all` hides logs that are visible otherwise. @amnk Could you try to repro this and see if this PR fixes the problem?

amnk commented:
@brikis98 my manual tests show that all 4 issues are resolved by this PR. 

> #1529 
We actually had that in integration tests, but during a switch to logrus I updated a test instead of validating what behavior is preferable 🤦 . So I've also changed that back.

brikis98 commented on shell/prompt.go:
```diff
@@ -12,9 +12,14 @@ import (
 	//
 	// See https://github.com/gruntwork-io/terragrunt/issues/1524 for additional context
+	if terragruntOptions.NonInteractive {
+		terragruntOptions.Logger.Infof("The non-interactive flag is set to true, so assuming 'yes' for all prompts")
```
Hm, logging this without the prompt itself doesn't make much sense, as the user will have no idea what the "yes" was in response to! It also creates noise in the logs. I think we'd be better off:

1. Check if `--non-interactive` is set
2. If it is, log the prompt and then this "assuming yes" message, both using `Debugf`.
3. If it isn't, write the prompt to `stderr` directly.

amnk replied:
@brikis98 Updated. I initially thought that displaying the prompt and skipping it with assume message does not make much sense, but from a user stand point it is probably more clear. 👍

brikis98 replied:
Roger, thx!

brikis98 reviewed (approved):
LGTM! I'll kick off tests now.

brikis98 commented:
Tests passed! Merging and release now. Thanks @amnk!

brikis98 commented:
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.28.6

dudicoco commented:
@amnk @brikis98 @yorinasub17 the logs are still very verbose after the recent changes with log-level info.

With log-level warn the logs verbosity is perfect, but then the error messages get lost. Is it possible to get the verbosity of warn with the addition of the error messages?

Thanks

amnk commented:
This patch was the first one in the series, because changing of logging level was a surprise for many people :) So yeah, I will be working on another PR(s) to make it actually less verbose while still printing relevant error messages. 

@dudicoco btw, can you share specific log messages you think are not needed with default output?

dudicoco commented:
Thanks @amnk.

The following log lines appear hundreds of times within the output and I believe are not necessary with the normal log level:
```
=> Module /Users/xxx/Desktop/Repositories/xxx/xxx/eu-central-1/prod/xxx/s3-bucket/ml (excluded: true, dependencies: [])
```

```
INFO[0072]                                               prefix=[/Users/xxx/Desktop/Repositories/xxx]
```

In particular the `INFO[xxxx]` log line is a new addition which we didn't see before the recent upgrades.

brikis98 commented:
> The following log lines appear hundreds of times within the output and I believe are not necessary with the normal log level:
> 
> ```
> => Module /Users/xxx/Desktop/Repositories/xxx/xxx/eu-central-1/prod/xxx/s3-bucket/ml (excluded: true, dependencies: [])
> ```

Hm, I believe these are necessary when we are interactively prompt the user if they want to run their `run-all` command on all these modules. @amnk These probably need to be bundled into the prompt message show to users, so that if `--non-interactive` is set, they are only logged at the `debug` level, but otherwise, show up as epxected. Not sure why they would appear hundreds of times though?

> 
> ```
> INFO[0072]                                               prefix=[/Users/xxx/Desktop/Repositories/xxx]
> ```

Not sure what this is. @amnk Could you investigate?

dudicoco commented:
@brikis98 when running `terragrunt plan-all` the `=> Module` log line appear per directory in the stack, so if the stack has hundreds of directories the log line appears hundreds of times.

 In addition to hiding these when `--non-interactive` which is relevant only to the `apply-all` command, can we hide them within the `plan-all` output as well?

Thanks

amnk commented:
> Not sure what this is. @amnk Could you investigate?

@brikis98 I have opened a PR to fix this here: https://github.com/gruntwork-io/terragrunt/pull/1568.

brikis98 commented:
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.28.8 should improve `run-all` log output a little bit. More changes coming later.

lorengordon commented:
@brikis98 fyi, 0.28.8 still doesn't have binaries posted to github releases...

yorinasub17 commented:
@lorengordon Should be available now. Sorry about that!
